self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryContracts.java

```diff
@@ -22,43 +22,378 @@
+        final boolean contractExists = contracts.stream()
+            .anyMatch(c -> c.contributor().contributorId() == contributorId
+                && c.project().projectId() == projectId);
+        if (!contractExists) {
+            final ProjectInternal project = findOrCreateProject(projectId);
+            final ContributorInternal contributor =
+                findOrCreateContributor(contributorId);
+            final ContractInternal contract =
+                new ContractInternal(project, contributor);
+            //store the contract
+            contracts.add(contract);
+            //link this contract to its project and contributor
+            project.addContract(contract);
+            contributor.addContract(contract);
+        }
+    }
+
+    /**
+     * Returns a project by id from the projects set.
+     * If the project is not found, a project will be created and added to set.
+     *
+     * @param projectId Project's
+     * @return Internal representation of a Project
+     */
+    private ProjectInternal findOrCreateProject(final int projectId) {
+        ProjectInternal project = projects.stream()
+            .filter(p -> p.projectId() == projectId)
+            .findFirst()
+            .orElse(null);
+        if (project == null) {
+            project = new ProjectInternal(projectId, this);
+            projects.add(project);
+        }
+        return project;
+    }
+
+    /**
+     * Returns a contributor by id from the contributors set.
+     * If the contributor is not found, a contributor will
+     * be created and added to set.
+     *
+     * @param cbId Contributor's id
+     * @return Internal representation of a Contributor
+     */
+    private ContributorInternal findOrCreateContributor(final int cbId) {
+        ContributorInternal contributor = contributors.stream()
+            .filter(c -> c.contributorId() == cbId)
+            .findFirst()
+            .orElse(null);
+        if (contributor == null) {
+            contributor = new ContributorInternal(cbId, this);
+            contributors.add(contributor);
+        }
+        return contributor;
+    }
+
+
+    /**
+     * In memory implementation of a {@link Project}.
+     * <br>
+     * Aids {@link InMemoryContracts}
+     * <br>
+     * Has capability to link a contract.
+     * See: {@link ProjectInternal#addContract(Contract)}.
+     */
+    private static final class ProjectInternal implements Project {
+
+        /**
+         * Project's id.
+         */
+        private final int id;
+        /**
+         * Global contracts. Access to {@link InMemoryContracts }
+         * Grants access to any arbitrary project's contracts using
+         * {@link Contracts#ofProject(int) from any point of the object graph.
+         */
+        private final Contracts rootContracts;
+
+        /**
+         * Contracts associated to this project.
+         */
+        private final Set<Contract> ownedContracts = new HashSet<>();
```

Comment:
@criske We don't really need this Set. The Project already has the ``rootContracts``, which should have an ``add`` method... but we can leave it as it is for now, until we implement ``Contracts.register``, or something.